packages/@aws-cdk/aws-servicecatalog/README2.md

```diff
@@ -0,0 +1,379 @@
+and the product will become visible within the portfolio side in both the Service Catalog console and AWS CLI.
+You can add a product to multiple portfolios depending on your organizational structure and how you would like to group access to products.
+
+```ts fixture=portfolio-product
+portfolio.addProduct(product);
+```
+
+## Tag Options
+
+TagOptions allow administrators to easily manage tags on provisioned products by providing a template for a selection of tags that end users choose from.
+TagOptions are created by specifying a tag key with a set of allowed values and can be associated with both portfolios and products.
+When launching a product, both the TagOptions associated with the product and the containing portfolio are made available.
+
+At the moment, TagOptions can only be deactivated in the console.
+
+```ts fixture=portfolio-product
+const tagOptionsForPortfolio = new servicecatalog.TagOptions(this, 'OrgTagOptions', {
+  allowedValuesForTags: {
+    Group: ['finance', 'engineering', 'marketing', 'research'],
+    CostCenter: ['01', '02','03'],
+  },
+});
+portfolio.associateTagOptions(tagOptionsForPortfolio);
+
+const tagOptionsForProduct = new servicecatalog.TagOptions(this, 'ProductTagOptions', {
+  allowedValuesForTags: {
+    Environment: ['dev', 'alpha', 'prod'],
+  },
+});
+product.associateTagOptions(tagOptionsForProduct);
+```
+
+## Constraints
+
+Constraints are governance gestures that you place on product-portfolio associations that allow you to manage minimal launch permissions, notifications, and other optional actions that end users can perform on products.
+Using the CDK, if you do not explicitly associate a product to a portfolio and add a constraint, it will automatically add an association for you.
+
+There are rules around how constraints are applied to portfolio-product associations.
+For example, you can only have a single "launch role" constraint applied to a portfolio-product association. 
+If a misconfigured constraint is added, `synth` will fail with an error message.
+
+Read more at [Service Catalog Constraints](https://docs.aws.amazon.com/servicecatalog/latest/adminguide/constraints.html).
+
+### Tag update constraint
+
+Tag update constraints allow or disallow end users to update tags on resources associated with an AWS Service Catalog product upon provisioning.
+By default, if a Tag Update constraint is not configured, tag updating is not permitted.
+If tag updating is allowed, then new tags associated with the product or portfolio will be applied to provisioned resources during a provisioned product update.
+
+```ts fixture=portfolio-product
+portfolio.addProduct(product);
+
+portfolio.constrainTagUpdates(product);
+```
+
+If you want to disable this feature later on, you can update it by setting the "allow" parameter to `false`:
+
+```ts fixture=portfolio-product
+// to disable tag updates:
+portfolio.constrainTagUpdates(product, {
+  allow: false,
+});
```

Comment:
The naming of this method is confusing to me. It's not clear if `allow` refers to allowing tag updates or allowing them to be constrained.

Reply:
+1, I would like to see better naming here

Reply:
Yeah, it's not the best name - however our actual service and functionality refers to these as constraints.  The underlying API invokes a `TagUpdateConstraint`.  While there is default behavior, since our customers are in large and regulated enterprises, they sometimes prefer explicit denies/allows, which also makes this peculiar since we cannot just have a boolean like method that creates the non default path and if you don't want it it just deletes.